src/googmodule.ts

```diff
@@ -209,19 +209,30 @@ function rewriteModuleExportsAssignment(expr: ts.ExpressionStatement) {
  * @return An array of statements if it converted, or null otherwise.
  */
 function rewriteCommaExpressions(expr: ts.ExpressionStatement): ts.Statement[]|null {
-  // Early exit if the outer statement isn't a comma statement.
-  if (!ts.isBinaryExpression(expr.expression)) return null;
-  if (expr.expression.operatorToken.kind !== ts.SyntaxKind.CommaToken) return null;
+  // There are two representation for comma expressions:
+  // 1) A "comma list" expression, where the subexpressions are in one array
+  if (expr.expression.kind === ts.SyntaxKind.CommaListExpression) {
+    const commaList = expr.expression as ts.CommaListExpression;
+    return commaList.elements.map(x => ts.setOriginalNode(ts.createExpressionStatement(x), x));
+  }
+
+  // or,
+  // 2) a tree of  "binary expressions" whose contents are comma operators
+  if (ts.isBinaryExpression(expr.expression) &&
+      expr.expression.operatorToken.kind === ts.SyntaxKind.CommaToken) {
+    return visitBinaryCommas(expr.expression)
+  }
 
   // Recursively visit comma-separated subexpressions, and collect them all as
   // separate expression statements.
-  return visit(expr.expression);
-  function visit(expr: ts.Expression): ts.Statement[] {
+  function visitBinaryCommas(expr: ts.Expression): ts.Statement[] {
```

Comment:
I guess technically we could end up with a binary expression containing a comma list expression in turn, but that's probably practically not going to happen, is it? You could consider adding a comment.

Reply:
I don't think TS produces cases with mixed comma structure, so I can't e.g. write a unit test, but I reworked the algorithm so it at least looks like it should work with mixed comma expression trees.

Reply:
I don't think TS ever produces "mixed comma trees" with both styles of expressions, so I can't e.g. write a good unit test, but I've shifted things around so it at least looks like it would work with a mixed comma tree.

Reply:
Sorry, I wasn't precise. I meant to say that we don't need to handle that problem, and a comment would have been enough. Doesn't hurt to cover it though.

Reply:
Now that I've refactored it, the code that does handles the pattern seems simpler anyway.